kubespawner/reflector.py

```diff
@@ -346,6 +355,9 @@ async def _watch_and_update(self):
                 self.log.debug("Cancelled watching %s", self.kind)
                 raise
             except Exception:
+                # ensure we request a valid resource version on retry,
+                # needed on 410 Gone errors
+                resource_version = "0"
```

Comment:
added this, should not be needed with the `NotOlderThan` resource_version_match but it probably makes sense to start from the beginning on errors.